lib/zclient.c

```diff
@@ -3341,6 +3449,22 @@ static int zclient_read(struct thread *thread)
 		break;
 	case ZEBRA_ERROR:
 		zclient_handle_error(command, zclient, length, vrf_id);
+		break;
+	case ZEBRA_OPAQUE_MESSAGE:
+		if (zclient->opaque_msg_handler)
+			(*zclient->opaque_msg_handler)(command, zclient, length,
+						       vrf_id);
+		break;
+	case ZEBRA_OPAQUE_REGISTER:
+		if (zclient->opaque_register_handler)
+			(*zclient->opaque_register_handler)(command, zclient,
+							    length, vrf_id);
+		break;
+	case ZEBRA_OPAQUE_UNREGISTER:
```

Comment:
This isn't supported yet, in this initial version. I wanted to propose the basic mechanism, and then let the actual uses help drive changes. There are some other optimizations or extensions that we've talked about too, so I do expect there will be some enhancements going forward.

Reply:
@rwestphal Regarding the re-synchro for e.g. BGP-LS, it was decided to use sub-type of ZAPI OPAQUE message. @mjstapp proposes with this PR a generic tansport of OPAQUE message between daemon. For what you are mention, the exchange of OPAQUE message is in the opposite direction. Let me explain:

- BGP or Pathd subscribe to the ZAPI OPAQUE message with sub-type Link-State-Update. IS-IS or OSPF send to Zebra Link State Update each time a new modification occurs.
- For Re-sync of Link-State, it is IS-IS or OSPF tat subscribe to the ZAPI OPAQUE message with sub-type Link-State-ReSync. When a dameon like BGP-LS or PATHd ask for this re-sync, it start by sending the Opaque Message with Link-State-Re-Sync sub-type and in turn, OSPF or IS-IS send back the whole Link State information. Here, we need certainly to specialise the sub-type (or add an option) to flood teh whole Link-State information to the daemon that request it instead of all deamon. So, perhaps, what we need with the OPAQUE ZAPI message is a flooding message and a direct message through a distinct OPAQUE ZAPI message of an option in the header.